### ISSUE TITLE
Validation: vkCmdClearAttachments validation not checking all parameters

### ISSUE THREAD

**Issue by pdaniell-nv** [KhronosGroup/Vulkan-LoaderAndValidationLayers#398]
In the spec for vkCmdClearAttachments (https://www.khronos.org/members/login/sandbox/vulkan/wsi/xhtml/vkspec.html#vkCmdClearAttachments) it lists several rules including:
- The rectangular region specified by a given element of pRects must be contained within the render area of the current render pass instance

Looking at the validation code for vkCmdClearAttachments in https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/master/layers/parameter_validation.cpp#L3429 it doesn't appear this rule and many of the others from the spec are being checked.

**Comment by lentinem** [comment]
There are many valid usage checks that are not in the layers atm. We do have a task for improving this and a fairly comprehensive list for what is and needs to be done. If you feel that a specific check is important compared to others feel free to post bugs like this but if not these will be addressed as part of that effort.

**Comment by pdaniell-nv** [comment]
I'm not sure about the relative priority of this one. I reported it only because it caught out a Vulkan developer even when had layers enabled.

**Comment by lentinem** [comment]
If it was seen in practice thats enough to bump the priority as a good chunk of the checks that need to be added will probably not be hit by the vast majority of people.

**Comment by pdaniell-nv** [comment]
Cool. Also issue #397 was seen in practice.

**Comment by yavn** [comment]
Hi, there's also a bug in `vkCmdClearAttachments()` related to checking attachment index in `VkClearAttachment`. The spec says:

> `colorAttachment` is only meaningful if VK_IMAGE_ASPECT_COLOR_BIT is set in aspectMask, in which case it is an index to the `pColorAttachments` array in the `VkSubpassDescription` structure of the current subpass which selects the color attachment to clear.

Note it says "in subpass description", so this line is wrong:

```
if (attachment->colorAttachment == pSD->pColorAttachments[i].attachment) {
```

The right hand side of condition is index in the render pass (create info), rather than the subpass. It should be compared to `i` instead.

**Comment by mark-lunarg** [comment]
This is presently in-work.

**Comment by mark-lunarg** [comment]
Fixed by PR #1385.
